src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp

```diff
@@ -2155,6 +2171,293 @@ class at_icc_start_frostwing_gauntlet : public AreaTriggerScript
+            {
+                case ACTION_PUTRICIDE_BUG_LEAP:
+                    DoCast(SPELL_RANDOM_LEAP);
+                    _events.ScheduleEvent(EVENT_END_LEAP, 3000);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            if (_events.ExecuteEvent() == EVENT_END_LEAP)
+            {
+                me->GetMotionMaster()->MoveRandom(10.0f);
+            }
+
+            if (!UpdateVictim())
+                return;
+
+            DoMeleeAttackIfReady();
+        }
+
+    private:
+        EventMap _events;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetIcecrownCitadelAI<npc_flesh_eating_insectAI>(creature);
+    }
+};
+
+class npc_putricide_trap : public CreatureScript
+{
+public:
+    npc_putricide_trap() : CreatureScript("npc_putricide_trap") { }
+
+    struct npc_putricide_trapAI : public ScriptedAI
+    {
+        npc_putricide_trapAI(Creature* creature) : ScriptedAI(creature)
+        {
+            _instance = creature->GetInstanceScript();
+        }
+
+        void DoAction(int32 action) override
+        {
+            switch (action)
+            {
+                case ACTION_PUTRICIDE_STALKERS:
+                    GetCreatureListWithEntryInGrid(stalkers, me, NPC_INVISIBLE_STALKER, 35.0f);
+                    _events.ScheduleEvent(EVENT_SUMMON_FLESH_EATING_BUG, 2000);
+                    break;
+                case ACTION_PUTRICIDE_TRAP:
+                    _events.ScheduleEvent(EVENT_PUTRICIDE_TRAP, 3000);
+                    break;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_PUTRICIDE_TRAP:
+                        DoCast(SPELL_GIANT_SWARM);
+                        _instance->SetData(DATA_PUTRICIDE_TRAP, IN_PROGRESS);
+                        _events.ScheduleEvent(EVENT_END_TRAP, 50000);
+                        break;
+                    case EVENT_SUMMON_FLESH_EATING_BUG:
+                        for (std::list<Creature*>::iterator itr = stalkers.begin(); itr != stalkers.end(); ++itr)
+                            (*itr)->AI()->DoCast(SPELL_SUMMON_PLAGUE_INSECT);
+                        if (_instance->GetData(DATA_PUTRICIDE_TRAP) == IN_PROGRESS)
+                            _events.ScheduleEvent(EVENT_SUMMON_FLESH_EATING_BUG, 2500);
+                        break;
+                    case EVENT_END_TRAP:
+                        me->RemoveAura(SPELL_GIANT_SWARM);
+                        me->CastStop();
+                        _instance->SetData(DATA_PUTRICIDE_TRAP, DONE);
+                        Map::PlayerList const &players = me->GetMap()->GetPlayers();
```

Comment:
I think neither of them is necessary, afaik aura gets removed on targets on owner dead.